### PR TITLE
Lint Extension Elements and Their Properties

### PR DESCRIPTION
**Pull request by philippfromme**
This pull request adds checks for extension elements and their properties and references (`bpmn:Error`, `bpmn:Message`) to the existing lint rules. These checks are implemented by composing checks that are reusable.

The following checks have been added:

| Element Type | Check |
|---|---|
| e.g. Service Task | `zeebe:TaskDefinition` with `zeebe:type` and `zeebe:retries` properties |
| Business Rule Task | `zeebe:CalledDecision` with `zeebe:decisionId` and `zeebe:resultVariable` properties |
| Call Activity | `zeebe:CalledElement` with `zeebe:processId` property |
| e.g. Service Task | `zeebe:LoopCharacteristics` with `zeebe:inputCollection`, `zeebe:outputCollection` and `zeebe:outputElement` properties |
| e.g. Receive Task | `zeebe:Subscription` with `zeebe:correlationKey` property |
| e.g. Error Boundary Event | `bpmn:messageRef` |
| e.g. Message Boundary Event | `bpmn:messageRef` |

Full list of errors can be found here: https://docs.google.com/spreadsheets/d/1AbFn2lXWPywmM1OaP7pVZUaAG0CglYv-2th2dbl5_k4/edit#gid=0

---
Closes https://github.com/camunda/bpmnlint-plugin-camunda-compat/issues/7
Related to https://github.com/bpmn-io/internal-docs/issues/430

### CODE REVIEW

**Comment by smbea** [comment]
Will have a look at this now

File: rules/camunda-cloud-1-0-checks.js [hunk 1]
```diff
@@ -29,69 +39,99 @@ module.exports = [
...
-    check: checkSome(
-      hasNoLoopCharacteristics,
-      hasLoopCharacteristicsOfType('bpmn:MultiInstanceLoopCharacteristics')
+    check: withTranslations(
+      checkEvery(
+        hasZeebeLoopCharacteristics,
+        hasZeebeTaskDefinition
+      ),
+      {
+        'Element of type <bpmn:ServiceTask> must have <zeebe:TaskDefinition> extension element': 'Must have implementation'
```
Comment (smbea) [inline rules/camunda-cloud-1-0-checks.js:107]: I'm confused here. When I create a ServiceTask, I get "Must have implementation", which isn't very specific. Is this a placeholder for the friendly messages?

Comment (philippfromme) [reply rules/camunda-cloud-1-0-checks.js:107]: This is just an example to verify that we can easily adjust the messages once we've come up with friendly error messages. So this is going to change. 😉

**Comment by philippfromme** [comment]
@smbea I'm making some last-minute adjustments as I am currently testing this end-to-end.

**Comment by philippfromme** [comment]
I've documented the different types of errors reported by the checks: https://github.com/camunda/bpmnlint-plugin-camunda-compat/pull/18/commits/6dae94e5aaa4081dde9bffb936222e237cccf304#diff-26fcc353c2a517aef63bf6a6be27b3892d084f3100b2b49756ee715ee37b2b7e

Ready to be reviewed. ✔️

File: rules/camunda-cloud-1-0-checks.js (hunk 1 above)
Comment (smbea) [reply rules/camunda-cloud-1-0-checks.js:107]: Ok makes sense

**Review by smbea — APPROVED** [review]
As far as I could tell, everything makes sense and is working fine

**Pull request merged**